src/main/kotlin/uk/gov/justice/digital/hmpps/welcometoprison/model/arrivals/ArrivalsService.kt

```diff
@@ -57,34 +59,46 @@ class ArrivalsService(
 
     val arrival = getArrival(moveId)
 
+    return if (!arrival.isCurrentPrisoner)
+      when (arrival.prisonNumber) {
+        null -> createAndAdmitOffender(confirmArrivalDetail, moveId)
+        else -> admitOffender(confirmArrivalDetail, moveId, arrival.prisonNumber)
+      } else
+      throw IllegalArgumentException("Feature not available.")
+  }
+
+  fun confirmReturnFromCourt(
+    moveId: String,
+    confirmCourtReturnRequest: ConfirmCourtReturnRequest
+  ): ConfirmCourtReturnResponse {
+
+    val arrival = getArrival(moveId)
+
     return if (arrival.isCurrentPrisoner)
       when (arrival.fromLocationType) {
-        LocationType.COURT -> transferInFromCourt(moveId, confirmArrivalDetail, arrival)
+        LocationType.COURT -> returnFromCourt(moveId, confirmCourtReturnRequest, arrival)
         else -> throw IllegalArgumentException("The arrival is known to NOMIS, has a current booking but is not from court. This scenario is not supported.")
       } else
-      when (arrival.prisonNumber) {
-        null -> createAndAdmitOffender(confirmArrivalDetail, moveId)
-        else -> admitOffender(confirmArrivalDetail, moveId, arrival.prisonNumber)
-      }
+      throw IllegalArgumentException("Feature not available.")
```

Comment:
can we make this more descriptive? 
`Confirming arrival of inactive prisoner: '${arrival.prisonNumber}' is not supported`

Reply:
done